### PR TITLE
Establish business logic unit tests and testing strategy foundation (Vitest, DDD/FSD)

### PR DESCRIPTION
**Pull request by lapidix**
### Overview

This project adopts a robust and scalable testing strategy focused on business logic validation for FSD/DDD/Clean Architecture. The current test coverage targets all core business logic layers and is designed for maintainability and future growth. UI tests (component/E2E) will be introduced in a separate phase.

### Key Points

- **Test Framework**
    
    - Uses Vitest and Testing Library as the main test stack
        
    - Dependencies: `vitest`, `@testing-library/react`, `@testing-library/jest-dom`, etc.
        
    - Global setup via `setup.ts` (includes console mocking for cleaner output)
        
- **Scope of Unit Tests**
    
    - Business logic layers: domain, service, hooks
        
    - Focuses on core domain rules, service orchestration, and custom hooks
        
    - Targets:
        
        - Entities: User, Post, Comment
            
        - Factories, mappers, repositories, services, hooks
            
    - Covers domain rules, validation, error handling, and edge cases
        
- **Test Utilities & Structure**
    
    - Centralized and standardized test data, helpers, and mocks
        
    - Improved reusability, readability, and consistency across the codebase
        
- **Configuration**
    
    - Test-related settings in `.gitignore`, `package.json`, and other config files
        
    - Scripts for running, watching, and coverage reporting via Vitest
        

### Motivation
 -  Build a robust, scalable test foundation for FSD/DDD/Clean Architecture codebase
 - Ensure business logic is fully covered to prevent regression and maintain code quality during refactoring and feature work
    

### Next Steps

- UI and E2E test coverage will be addressed in future updates
    
- Continuous improvement of test coverage and test practices

### CODE REVIEW

**Comment by lapidix** [comment]
@gemini-code-assist review

File: src/features/comment/__tests__/services/comment.service.test.ts [hunk 1]
```diff
@@ -0,0 +1,942 @@
...
+
+      // When: Get comment by ID
+      // Then: Should re-throw the same BaseError
+      await expect(commentService.getCommentById(commentId)).rejects.toThrow(
+        baseError
+      );
+    });
+  });
+
+  describe("addComment", () => {
```
[the remark this answers is not in the document]
Comment (lapidix) [reply src/features/comment/__tests__/services/comment.service.test.ts:148]: @gemini-code-assist This is a sample code, assuming that the client only has the userId.

**Comment by lapidix** [comment]
@gemini-code-assist review

File: src/entities/comment/infrastructure/api/comment.adapter.ts [hunk 2]
```diff
@@ -1,42 +1,30 @@
...
-      };
-    }
+    const response = await apiClient.get<Pagination<CommentDto>>(
+      `/posts/${postId}/comments`
+    );
+    return response.data;
   },
 
-  getById: async (id: string): Promise<CommentDto | null> => {
+  getById: async (id: string): Promise<CommentDto> => {
```
[the remark this answers is not in the document]
Comment (lapidix) [reply src/entities/comment/infrastructure/api/comment.adapter.ts:13]: @gemini-code-assist All of the review comments below have already been addressed, but it seems the latest code was not reflected at the time of review.

**Comment by lapidix** [comment]
@gemini-code-assist review
